packages/commerce-server/src/format-prices-for-product.ts

```diff
@@ -184,12 +176,18 @@ export async function formatPricesForProduct(
   const pppConditionsMet =
     quantity === 1 && pppDiscountPercent > 0 && appliedMerchantCouponLessThanPPP
 
+  const hasPurchaseWithPPP = userPurchases.some(
+    (purchase) =>
+      purchase.status === 'Restricted' &&
+      upgradeFromPurchase &&
+      purchase.productId === upgradeFromPurchase?.productId,
+  )
```

Comment:
this should really only be relevant if they are upgrading a product with PPP restrictions, but we could automatically apply it across the board if they have any PPP too. 

That would be making the assumption that they would want PPP applied to any/all purchases, which I feel is safe?

If that's the case we could remove the product id comparison.